python/oneseismic/scan/tests/test_scan.py

```diff
@@ -40,3 +42,65 @@ def test_regular_intervals(inlines, crosslines):
         seg.scan_trace_header(header)
 
     assert len(seg.key2s) == crosslines
+
+@given(
+    integers(min_value = 2, max_value = 15),
+    integers(min_value = 2, max_value = 15),
+    floats(min_value = 1, max_value = 10),
+    floats(min_value = 1, max_value = 10),
+    floats(min_value = -math.pi, max_value = math.pi),
+    floats(min_value = -10, max_value = 10),
+    floats(min_value = -10, max_value = 10)
+)
+def test_rot_step(
+        inlines,
+        crosslines,
+        ilstep,
+        xlstep,
+        rotation,
+        offsetx,
+        offsety
+):
+    ilstepx = ilstep * math.cos(rotation)
+    ilstepy = ilstep * math.sin(rotation)
+    xlstepx = xlstep * math.cos(rotation)
+    xlstepy = xlstep * math.sin(rotation)
+
+    headers = [
+        {
+            int(segyio.su.iline): big_endian(i),
+            int(segyio.su.xline): big_endian(x),
+            int(segyio.su.cdpx): i * ilstepx + x * xlstepx + offsetx,
+            int(segyio.su.cdpy): i * ilstepy + x * xlstepy + offsety,
```

Comment:
Yes, there was an error in the generation of the line step variables, leading to some strange cdp values. 

The test does still verify that the scanner returns the inline and crossline steps that was used to construct the cdp header values, so that gives me a lot of confidence that it does work as expected.

If I was creating the test data by hand, these are still the calculations I would do, so it does not add to correctness. And I don't think that looking at these values would give a better insight into how it works.

Perhaps something could be done with the variable names to make it more clear?

Reply:
1. Exactly. 
Hypothesis uses random values. There was an error and hypothesis didn't fail. Why would I believe it? It's good as a supporting test or in cases where you can't yourself think of anything that can fail.
I also can't imagine how it can be debugged.
With supplying an example you are forced to find useful, simple and human-readable values that will illustrate the problem. You can draw a pretty picture. You can think about things you tried to solve.
So I think that one explicit example would do you some good.

2. Data generation gives additional test complexity to the code complexity. I want to look at the test, read supporing comments if necessary and quickly see why this test correctly tests what it is supposed to.
So while I am fine with this hypothesis test staying, I need to see why the values you supply are the correct ones. You think too well of ~me~ someone who will read the code to figure it out just from variable names.

3. Current test setup doesn't even have missing points, which was important feature you tried to combat. Also all values are completely uniform, which gives the impression this is intended.